File: platform/app/models/content.py
"""Content domain model (unified from ContentV3.js and Content.js)."""

from __future__ import annotations

from datetime import datetime

from bson import ObjectId
from pydantic import BaseModel, ConfigDict, Field


class TextContent(BaseModel):
    """Bilingual text with optional audio URL."""

    english: str
    local: str = ""
    audio_url: str = ""

    model_config = ConfigDict(populate_by_name=True)


class AudioContent(BaseModel):
    """A single audio segment within a content item."""

    description: str = ""
    audio_url: str = ""
    duration_seconds: float | None = None

    model_config = ConfigDict(populate_by_name=True)


class Content(BaseModel):
    """Unified content document covering both contentsV2 and contentsV3 collections.

    Use the ``version`` field to distinguish:
      - "v2" : legacy (contentsV2 collection, flat string fields)
      - "v3" : current (contentsV3 collection, structured TextContent fields)
    """

    model_config = ConfigDict(populate_by_name=True)

    id: str | None = Field(None, alias="_id")
    version: str = "v3"
    tenant_id: str | None = None
    description: str = ""
    type: str
    language: str
    # V3 structured title/theme
    title: TextContent | None = None
    theme: TextContent | None = None
    # V2 flat title/theme fields
    title_text: str | None = None
    local_title: str | None = None
    title_audio: str | None = None
    theme_text: str | None = None
    local_theme: str | None = None
    theme_audio: str | None = None
    # Audio content (v3)
    audio_content: list[AudioContent] = Field(default_factory=list)
    # Flags
    school_id: str | None = None
    created_by: str = ""
    is_pull_model: bool = False
    is_teacher_app: bool = False
    is_processed: bool = False
    is_deleted: bool = False
    creation_time: int = -1
    created_at: datetime | None = None
    updated_at: datetime | None = None

    @classmethod
    def from_mongo(cls, doc: dict) -> Content:
        if doc is None:
            return None  # type: ignore[return-value]
        d = {k: str(v) if isinstance(v, ObjectId) else v for k, v in doc.items()}
        return cls.model_validate(d)

File: Teacher-App/app/src/main/java/com/example/seeds/model/Content.kt
package com.example.seeds.model

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.parcelize.Parcelize
import se.ansman.kotshi.JsonSerializable

@JsonSerializable
@Parcelize
data class Content(
    @Json(name = "id")
    val _id: String,
    val type: String,
    val description: String? = null,
    val language: String,
    val title: LocalizedContent?,       // JSON object
    val theme: LocalizedContent?,
    @Json(name = "audio_content")
    val audioContent: List<AudioContent> = emptyList(),
    @Json(name = "is_pull_model")
    val isPullModel: Boolean = false,
    @Json(name = "is_teacher_app")
    val isTeacherApp: Boolean = false,
    @Json(name = "created_by")
    val createdBy: String = "",
    val creation_time: Long = 0L,
    @Json(name = "is_deleted")
    val isDeleted: Boolean = false
) : Parcelable {
    val id: String get() = _id
    val titleText: String get() = title?.english ?: "Unknown Title"
    val themeText: String get() = theme?.english ?: "Unknown Theme"
}


@JsonSerializable
@Parcelize
data class LocalizedContent(
    val english: String? = null,
    val local: String? = null,
    @Json(name = "audio_url")
    val audioUrl: String? = null
) : Parcelable

@JsonSerializable
@Parcelize
data class AudioContent(
    val description: String,
    @Json(name = "audio_url")
    val audioUrl: String?
) : Parcelable
